src/user-statuses/userStatusesReducer.js

```diff
@@ -0,0 +1,95 @@
+/* @flow strict-local */
+import Immutable from 'immutable';
+
+import type { UserStatusUpdate } from '../api/modelTypes';
+import { makeUserId } from '../api/idTypes';
+import objectEntries from '../utils/objectEntries';
+import type { UserStatusesState, UserStatus, PerAccountApplicableAction } from '../types';
+import {
+  LOGOUT,
+  LOGIN_SUCCESS,
+  ACCOUNT_SWITCH,
+  REGISTER_COMPLETE,
+  EVENT_USER_REMOVE,
+  EVENT_USER_STATUS_UPDATE,
+} from '../actionConstants';
+
+const initialState: UserStatusesState = Immutable.Map();
+
+/**
+ * The canonical default, "unset" user status.
+ *
+ * This is the user-status you have if you've just created your account and
+ * never interacted with the feature.
+ *
+ * It's effectively the user-status you have if you're on an old server that
+ * doesn't support user statuses.
+ */
+// PRIVATE: Only to be used in this model's code.
+export const kUserStatusZero: UserStatus = {
+  away: false,
+  status_text: null,
+  status_emoji: null,
+};
+
+function updateUserStatus(
+  status: UserStatus,
+  update: $ReadOnly<{ ...UserStatusUpdate, ... }>,
+): UserStatus {
+  const { away, status_text, emoji_name, emoji_code, reaction_type } = update;
+  return {
+    away: away ?? status.away,
+    status_text: status_text ?? status.status_text,
```

Comment:
I believe this means that if an update comes in with `status_text: ''`, meaning there is no status text set, we'll get `status_text: ''` in our internal state, whereas the value the UserStatus jsdoc says should represent that is `status_text: null`. IOW, this logic and that type definition disagree on whether to use `null` or `''`.

I agree with the reasoning at https://github.com/zulip/zulip-mobile/pull/5224#discussion_r804345618 that `null` is nice here because Flow can distinguish it from strings. To reflect that choice here, I think it just needs to get a bit more complicated in a similar way to `status_emoji` below:
```suggestion
    status_text: status_text != null ? status_text === '' ? null : status_text : status.status_text,
```

Reply:
This is probably also a good prompt to add some more test cases, in particular covering both updating from the zero value to something else and vice versa.

Hmm, also covering status emoji.